src/app/services/actions/query-action-creators.ts

```diff
@@ -0,0 +1,26 @@
+import { IAction } from '../../../types/query-runner';
+import { QUERY_GRAPH_ERROR, QUERY_GRAPH_SUCCESS } from '../constants';
+
+export function queryResponse(response: string): IAction {
+  return {
+    type: QUERY_GRAPH_SUCCESS,
+    response,
+  };
+}
+
+export function queryResponseError(response: string): IAction {
+  return {
+    type: QUERY_GRAPH_ERROR,
+    response,
+  };
+}
+
+export function runQuery(url: string): Function {
+  const headers = { Authorization: 'Bearer {token:https://graph.microsoft.com/}' };
+
+  return (dispatch: Function) => {
+    return fetch(`https://proxy.apisandbox.msdn.microsoft.com/svc?url=${url}`, { headers })
+      .then((response) => response.json(), (error) => queryResponseError(error))
```

Comment:
How does queryResponseError get dispatched?  I only see the dispatch for success.

Reply:
I left out error handling for the prototype. Should have communicated that clearly. Should I add error handling?